app.js

```diff
@@ -70,18 +70,58 @@ bot.on('text', (ctx) => {
             if(textMsg.toLowerCase().search('mybatchmates') >= 0) {
                 //get my batchmates
                 dbo.collection(config.get('mongoCollections.users')).findOne({"from.id" : fromId}, function(error, user) {
-                    var reply = "Here are your batchmates : \n";
-                    dbo.collection(config.get('mongoCollections.users')).find({"batch" : user.batch}).forEach(function(batchmate) {
+                    var reply = `Here are your batchmates : \n`;
+                    dbo.collection(config.get('mongoCollections.users')).find({"batch" : user.batch}).toArray( function(err , batchmates) {
+                        batchmates.map(batchmate => {
+                            console.log(batchmate)
+                            reply += batchmate.from.first_name ;
+                            if( batchmate.from.last_name) {
+                                reply +=  `${batchmate.from.last_name} \n`;
+                            } 
+                        })
+                        utility.sendMessage(ctx, reply, 'search', 'private'); 
+                    })
+                })
+            } else if(textMsg.toLowerCase().search('batch_') >= 0) {
+                //get batchmates of given year { pattern is /batch_YYYY }
+                var year = textMsg.substring(textMsg.length - 4, textMsg.length); 
+                console.log(year)
+                var reply = `Here are batchmates from ${year} : \n`;
+                dbo.collection(config.get('mongoCollections.users')).find({"batch" : year}).toArray( function(err , batchmates) {
+                    batchmates.map(batchmate => {
                         reply += batchmate.from.first_name ;
                         if( batchmate.from.last_name) {
-                            reply += " " + batchmate.from.last_name;
+                            reply +=  `${batchmate.from.last_name} \n`;
                         } 
-                        reply += "\n";
-                        }, function(latNull) {
-                            utility.sendMessage(ctx, reply, 'search', 'private'); 
-                    }) 
+                    })
+                    if(!batchmates.length){
+                        reply = `No one from ${year} batch is registered.`;
+                    }
+                    utility.sendMessage(ctx, reply, 'search', 'private'); 
+                })
+                
+            } else if(textMsg.toLowerCase().search('from_city_') >= 0) {
+                // pattern is /from_city_CITYNAME
+                var tempMsg = textMsg.split("_"); 
+                var city = tempMsg[tempMsg.length-1];
+                console.log(city)
+                var reply = `Here are peoples from ${city} : \n`;
+                dbo.collection(config.get('mongoCollections.users')).find({"location" : new RegExp(city)}).toArray( function(err , users) {
```

Comment:
I have never tried new RegExp(), $regex is there in mongo, but its fine, if you have tested it and its result are fine.

Reply:
@saumye1 Yes I have tested it and it gives the required result.

Reply:
@nkkumawat what's the benefit of using RegExp for the batch year?

Reply:
@devender-yadav if someone type /batch_98 instead of /batch_1998 then the bot will show required results.

Reply:
@nkkumawat sounds good. Adding these comments with the code itself is always good :)